lib/srv/desktop/rdp/rdpclient/src/client/global.rs

```diff
@@ -0,0 +1,94 @@
+//! This module contains static structures which are used in common by all
+//! desktop sessions on a given windows_desktop_service.
+//!
+//! ## Constraints
+//!
+//! The primary constraint for maintainers to keep in mind is that these
+//! structures can in effect be accessed by multiple threads at any given
+//! time via Go. Therefore, typical Rust concurrency semantics must be
+//! carefully observed (the compiler will not necessarily catch violations).
+//!
+//! In practice this primarily means ensuring that such global, static
+//! structures are [`Send`] + [`Sync`] and are only mutated when locked.
+
+use super::{ClientFunction, ClientHandle};
+use crate::{CGOErrCode, CgoHandle};
+use parking_lot::RwLock;
```

Comment:
@probakowski why do we prefer `parking_lot` here vs `std::sync::RwLock`?

Reply:
It's dubbed as more performant version than std, figured out it's worth using it since we access it frequently